src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorNodeWriter.cs

```diff
@@ -49,5 +53,169 @@ public sealed override void WriteCSharpCodeAttributeValue(CodeRenderingContext c
+                {
+                    writer.Write(", ");
+                    writer.Write(parameters[i].typeName);
+                    writer.Write(" ");
+                    writer.Write(parameters[i].parameterName);
+                }
+
+                if (i < parameters.Count - 1)
+                {
+                    writer.Write(", ");
+                }
+            }
+
+            writer.Write(")");
+            writer.WriteLine();
+
+            writer.WriteLine("{");
+
+            // builder.OpenComponent<TComponent>(42);
+            context.CodeWriter.Write("builder");
+            context.CodeWriter.Write(".");
+            context.CodeWriter.Write(BlazorApi.RenderTreeBuilder.OpenComponent);
+            context.CodeWriter.Write("<");
+            context.CodeWriter.Write(node.Component.TypeName);
+            context.CodeWriter.Write(">(");
+            context.CodeWriter.Write("seq");
+            context.CodeWriter.Write(");");
+            context.CodeWriter.WriteLine();
+
+            var index = 0;
+            foreach (var attribute in node.Component.Attributes)
+            {
+                context.CodeWriter.WriteStartInstanceMethodInvocation("builder", BlazorApi.RenderTreeBuilder.AddAttribute);
+                context.CodeWriter.Write(parameters[index].seqName);
+                context.CodeWriter.Write(", ");
+
+                context.CodeWriter.Write($"\"{attribute.AttributeName}\"");
+                context.CodeWriter.Write(", ");
+
+                context.CodeWriter.Write(parameters[index].parameterName);
+                context.CodeWriter.WriteEndMethodInvocation();
+
+                index++;
+            }
+
+            foreach (var childContent in node.Component.ChildContents)
+            {
+                context.CodeWriter.WriteStartInstanceMethodInvocation("builder", BlazorApi.RenderTreeBuilder.AddAttribute);
+                context.CodeWriter.Write(parameters[index].seqName);
+                context.CodeWriter.Write(", ");
+
+                context.CodeWriter.Write($"\"{childContent.AttributeName}\"");
+                context.CodeWriter.Write(", ");
+
+                context.CodeWriter.Write(parameters[index].parameterName);
+                context.CodeWriter.WriteEndMethodInvocation();
+
+                index++;
+            }
+
+            foreach (var capture in node.Component.Captures)
+            {
+                context.CodeWriter.WriteStartInstanceMethodInvocation("builder", capture.IsComponentCapture ? BlazorApi.RenderTreeBuilder.AddComponentReferenceCapture : BlazorApi.RenderTreeBuilder.AddElementReferenceCapture);
+                context.CodeWriter.Write(parameters[index].seqName);
+                context.CodeWriter.Write(", ");
+
+                var cast = capture.IsComponentCapture ? $"({capture.ComponentCaptureTypeName})" : string.Empty;
+                context.CodeWriter.Write($"(__value) => {{ {parameters[index].parameterName}({cast}__value); }}");
```

Comment:
It's the latter. We need to pass an `Action<object>` into the builder, but we happen to have an `Action<SomethingSpecific>`. We have an `Action<SomethingSpecific>` because we can't put the cast inside the lambda that gets passed to the thunk.